### DISCUSSION TITLE
Is there a way to add Makefiles as root_marker?

### DISCUSSION THREAD

**Discussion by MikeV2000** [nvim-java/nvim-java#119] [Q&A]
I would like to write a java project using make as my build tool. I like how gradle and maven projects are automatically configured. Is it possible to have the same configuration of make projects? Even a .git folder as root_marker would probably serve the purpose. 

But maybe I am just doing something wrong and that is already working. I am getting the warning "Main.java is a non-project file, only syntax errors are reported" and there is some functionality lacking.

**Comment by s1n7ax** [comment]
.git is already there as a root marker. Following properties are available for configuration. You can pass the changes to the setup function.

https://github.com/nvim-java/nvim-java/blob/main/lua/java/config.lua
